app/models/listed_taxon.rb

```diff
@@ -488,6 +490,41 @@ def trickle_down_establishment_means(options = {})
     ActiveRecord::Base.connection.execute(sql)
   end
   
+  def is_atlased?
+    if list.is_a?(CheckList) && list.is_default?
+      if atlas = Atlas.where(taxon_id: taxon_id).first
+        if atlas.places.map(&:id).include? place_id
+          return true
+        end
+      end
+    end
+    return false
+  end
+  
+  def log_create_in_atlas
+    if is_atlased?
+      atlas = Atlas.where(taxon_id: taxon_id).first
+      AtlasAlteration.create(
+        atlas_id: atlas.id,
+        user_id: user_id,
+        place_id: place_id,
+        action: "created"
+      )
+    end
+  end
+  
+  def log_destroy_in_atlas
+    if is_atlased?
+      atlas = Atlas.where(taxon_id: taxon_id).first
+      AtlasAlteration.create(
+        atlas_id: atlas.id,
+        user_id: user_id,
```

Comment:
This would leave a record stating that the person who *created* the listed taxon then destroyed it, which might not be the case since other people (like curators and admins) can destroy listed taxa, and listed taxa can be automatically destroyed. I think you need a way to pass down the `current_user` from the controller, perhaps using the `updater` relationship, though you should confirm that `updater` gets set on destruction, b/c it probably doesn't (why would you store that if you're going to delete the record).